src/bin/common/solving_loop.ml

```diff
@@ -354,6 +408,23 @@ let main () =
     State.create_key ~pipe:"" "is_incremental"
   in
 
+  (* Before each query, we push the current environment. This allows to keep a
+     fresh one for the next assertions. *)
+  let internal_push st =
+    let module Api = (val (DO.SatSolverModule.get st)) in
+    Steps.apply_without_step_limit (fun _ -> Api.FE.push 1 Api.env);
+    st
+  in
+
+  (* The pop corresponding to the previous push. It is applied everytime the
+     mode goes from Sat/Unsat to Assert. *)
+  let internal_pop st =
+    assert (List.mem (DO.Mode.get st) [Sat; Unsat]);
+    let module Api = (val (DO.SatSolverModule.get st)) in
+    Steps.apply_without_step_limit (fun _ -> Api.FE.pop 1 Api.env);
```

Comment:
Do we need to remove the step limit here? I think it was not done previously for `pop`.

Reply:
I had an issue when testing the push/pop test in the smtlib tests, I can retry to check if this was indeed triggered by this pop or not.